### PR TITLE
New Resource `azurerm_network_manager_scope_connection`

### PR DESCRIPTION
**Pull request by teowa**
Notes:

- This resource is the `Network manager connection` metioned in [cross-tenant-connection](https://learn.microsoft.com/en-us/azure/virtual-network-manager/concept-cross-tenant#cross-tenant-connection) doc,  testcases of this resource uses a single tenant due to test limitations.
- swagger: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/networkManagerScopeConnection.json

Test:
```
TF_ACC=1 go test -v ./internal/services/network -test.run=TestAccNetworkManager -timeout 1440m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccNetworkManager
=== RUN   TestAccNetworkManager/Manager
=== RUN   TestAccNetworkManager/Manager/requiresImport
=== RUN   TestAccNetworkManager/Manager/basic
=== RUN   TestAccNetworkManager/Manager/complete
=== RUN   TestAccNetworkManager/Manager/update
=== RUN   TestAccNetworkManager/ScopeConnection
=== RUN   TestAccNetworkManager/ScopeConnection/basic
=== RUN   TestAccNetworkManager/ScopeConnection/complete
=== RUN   TestAccNetworkManager/ScopeConnection/update
=== RUN   TestAccNetworkManager/ScopeConnection/requiresImport
--- PASS: TestAccNetworkManager (1904.83s)
    --- PASS: TestAccNetworkManager/Manager (988.38s)
        --- PASS: TestAccNetworkManager/Manager/requiresImport (242.18s)
        --- PASS: TestAccNetworkManager/Manager/basic (147.75s)
        --- PASS: TestAccNetworkManager/Manager/complete (157.65s)
        --- PASS: TestAccNetworkManager/Manager/update (440.80s)
    --- PASS: TestAccNetworkManager/ScopeConnection (916.45s)
        --- PASS: TestAccNetworkManager/ScopeConnection/basic (221.58s)
        --- PASS: TestAccNetworkManager/ScopeConnection/complete (169.06s)
        --- PASS: TestAccNetworkManager/ScopeConnection/update (282.09s)
        --- PASS: TestAccNetworkManager/ScopeConnection/requiresImport (243.71s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/network       1906.038s

```

### CODE REVIEW

**Review by katbyte — CHANGES_REQUESTED** [review]
test failures:
```
 Error: creating Network Manager: (Name "acctest-networkmanager-221211222359842005" / Resource Group "acctestRG-nm-221211222359842005"): network.ManagersClient#CreateOrUpdate: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="BadRequest" Message="Cannot have two Network Managers applied to the same object overlapping scope accesses. Scope id: /subscriptions/*******. Network Manager Id: /subscriptions/*******/resourceGroups/acctestRG-nm-221211222359842005/providers/Microsoft.Network/networkManagers/acctest-networkmanager-221211222359842005. Overlapping accesses: SecurityAdmin"
```

**Comment by teowa** [comment]
> 

https://github.com/hashicorp/terraform-provider-azurerm/pull/19593#issuecomment-1345977216

File: internal/services/network/network_manager_scope_connection_resource.go [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.NetworkManagerID,
+		},
+
+		"resource_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
```
Comment (tombuildsstuff) [inline internal/services/network/network_manager_scope_connection_resource.go:62]: we don't expose `resource_id` - what is this, a scope?

Comment (teowa) [reply internal/services/network/network_manager_scope_connection_resource.go]: modifies it to `target_scope_id`, it should be either Subscription ID or Management Group ID.

**Comment by teowa** [comment]
Provisioning two Network Manager using one subscription as scope leads to an error, to avoid test failure, temporary close this PR. 

Please review #19593 first.

**Comment by teowa** [comment]
image

**Review by katbyte — APPROVED** [review]
LGTM 🌵

**Pull request merged**